File: 2020W53/yurihan/두개뽑아서더하기.swift
// 두 개 뽑아서 더하기
// https://programmers.co.kr/learn/courses/30/lessons/68644

import Foundation

func solution(_ numbers:[Int]) -> [Int] {
    var ret: Set<Int> = []
    for i in 0..<(numbers.count-1)
    {
        for j in (i+1)..<(numbers.count)
        {
            ret.insert(numbers[i]+numbers[j])
        }
    }
    return Array<Int>(ret).sorted()
}

print(solution([2,1,3,4,1])) // [2,3,4,5,6,7]
print(solution([5,0,2,7])) // [2,5,7,9,12]

File: 2020W53/JH/Programmers/두개뽑아서더하기.java
//두개 뽑아서 더하기
//https://programmers.co.kr/learn/courses/30/lessons/68644
import java.util.Arrays;
import java.util.HashSet;
import java.util.Iterator;

public class Main {
    public static void main(String[] args) {
        int[] numbers = {5,0,2,7};

        System.out.println(Arrays.toString(solution(numbers)));
    }

    public static int[] solution(int[] numbers) {


        HashSet<Integer> tempSet = new HashSet<>();

        for(int i=0; i<numbers.length; i++) {
            for(int j=0; j< numbers.length; j++) {
                if(i == j) continue;
                tempSet.add(numbers[i] + numbers[j]);
            }
        }

        Iterator<Integer> iterator = tempSet.iterator();
        int[] answer = new int[tempSet.size()];

        int count = 0;

        while (iterator.hasNext()) {
            answer[count] = iterator.next();
            count++;
        }

        Arrays.sort(answer);

        return answer;
    }

}
